### PR TITLE
feat: Add includePackages option

### LINKED ISSUE #541: Enhancement: Add "includePackages" option
I am creating a hybrid android/ios app using [capacitor](https://capacitorjs.com/). The web part is built with webpack and uses webpack-license-plugin. But because of capacitor's structure, some dependencies like `@capacitor/android` are not imported in the web part, but used by the native wrapper around it. Because they are not imported in the web part, webpack-license-plugin cannot see these dependencies.

As a workaround I am using the following code to collect the licenses for a predefined list of packages:
```typescript
import LicenseMetaAggregator from 'webpack-license-plugin/dist/LicenseMetaAggregator';
import WebpackFileSystem from 'webpack-license-plugin/dist/WebpackFileSystem';
import fs from 'fs';
import PackageJsonReader from 'webpack-license-plugin/dist/PackageJsonReader';
import path from 'path';
import unacceptableLicenseTest from './unacceptableLicenseTest';

export const additionalPackages = async () => {
  const fileSystem = new WebpackFileSystem(fs);
  const alertAggregator = {
    addError: (message: string) => {
      throw new Error(message);
    },
    addWarning: () => {},
    flushAlerts: () => {}
  };
  const options = {
    additionalFiles: {},
    licenseOverrides: {},
    outputFilename: '',
    replenishDefaultLicenseTexts: true,
    unacceptableLicenseTest,
    excludedPackageTest: () => false
  };
  const licenseMetaAggregator = new LicenseMetaAggregator(fileSystem, alertAggregator, options, new PackageJsonReader(fileSystem));
  const modules = ['@capacitor/android', '@capacitor/ios'].map(m => path.dirname(require.resolve(`${m}/package.json`)));
  return await licenseMetaAggregator.aggregateMeta(modules);
};
```
The packages retuned by this function are then merged with the packages from webpack-license-plugin using the `additionalFiles` option:
```typescript
new LicensePlugin({
  additionalFiles: {
    'oss-licenses.json': async (packages) => {
      return JSON.stringify([...packages, ...(await additionalPackages())]);
    }
  }.
  replenishDefaultLicenseTexts: true,
  unacceptableLicenseTest
})
```

This works fine (it extracts the licenses correctly and throws errors for unacceptable licenses), but except for the moduleDirs source it is basicly the same code as inside the webpack-license-plugin. Therefore I would like to propose an additional option `includePackages` that allows to define a list of packages whose licenses should always be included in the license output.

### PR DESCRIPTION
**Pull request by WIStudent**
Closes #541 

I took a slightly different approach: Instead of of trying to resolve the package locations inside the webpack plugin, `includePackages` should contain a function that returns the absolute paths of the packages. This gives users more flexibility when including additional packages. They can directly link to the package inside their node_modules folder
```typescript
{
  includePackages: () => [
    // Assuming this config and node_modules are in the same directory
    path.join(__dirname, 'node_modules/some-package')
  ]
}
```
or use node to resolve the package location
```typescript
{
  includePackages: () => [
    path.dirname(require.resolve('some-package/package.json'))
  ]
}
```
This makes it even possible to point to packages inside other projects (might be relevant for some monorepo setups).

### CODE REVIEW

File: src/LicenseFileWriter.ts [hunk 1]
```diff
@@ -16,8 +16,9 @@ export default class LicenseFileWriter {
     options: IPluginOptions
   ): Promise<void> {
     const moduleDirs = this.getModuleDirs(filenames)
+    const includePackages = await options.includePackages();
     const licenseMeta = await this.licenseMetaAggregator.aggregateMeta(
-      moduleDirs
+      [...moduleDirs, ...includePackages]
```
Comment (codepunkt) [inline src/LicenseFileWriter.ts:21]: should we de-duplicate here to not read a folder/package.json twice?

File: test/e2e/index.test.ts [hunk 2]
```diff
@@ -180,6 +180,30 @@ describe('end to end', () => {
           }
         )
       })
+
+      test('output matches snapshot when packages have been included', done => {
+        build(
+          new WebpackLicensePlugin({
+            includePackages: () => [
+              resolve(__dirname, '../../node_modules/jest')
```
Comment (codepunkt) [inline test/e2e/index.test.ts:188]: nice 👍

**Review by codepunkt — COMMENTED** [review]
Awesome PR!

Besides the possible de-duplication of package folders, I was thinking about edge-cases like including a package version via `includePackage` twice, once because it's included in the build result, the second time because it's included in `includePackage`. If this points to the same directory, de-duplication where i made the code comment would resolve this. If the `includePackage` points to another directory that also has the same version of a module that's already included in the build, we have to ensure that de-duplication is not done on a later level because we'd always want to include both definitions (even though they seemingly point to the same package/version combination) as it might be possible that they differ in license regardless.

What do you think?

File: src/LicenseFileWriter.ts (hunk 1 above)
Comment (WIStudent) [reply src/LicenseFileWriter.ts:21]: I was not sure if deduplication is necessary here. There seems to happen some deduplication already in aggregateMeta, because when I tried the e2e test with
```typescript
includePackages: () => [
    resolve(__dirname, 'example/node_modules/react')
  ]
```
the resulting json contained the react depdencency only once. But if we want to avoid the duplicate read we can also deduplicate here:

```typescript
[...new Set([...moduleDirs, ...includePackages])]
```

As for the issue that two packages `folderA/node_modules/some-package` and `folderB/node_modules/some-package` might have the same name and version but different licenses: I think that's very unlikely. Npm guarantees that the name and version combination is unique and does not allow to replace an already published version. The only way this could happen is if one of the packages is a local package or from a private registry and someone made the mistake of using the same name/version as an already existing package for his local/private package.

Comment (codepunkt) [reply src/LicenseFileWriter.ts:21]: I agree that this is unlikely and would be fine with listing such a package twice if this happens. I like the deduplication via `Set`!

**Comment by codepunkt** [comment]
Btw, feel free to add yourself to `package.json` contributors 😁

**Review by codepunkt — APPROVED** [review]

**Comment by codepunkt** [comment]
released as `4.2.0`
